### PR TITLE
fix(rest): Added endpoint for mysubscriptions for component

### PR DESCRIPTION
**Pull request by nikkuma7**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: Subscription list for user in component (MY SUBSCRIPTIONS)

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Testing was Successful.

**Comment by rudra-superrr** [comment]
Code looks good.

File: rest/resource-server/src/docs/asciidoc/components.adoc [hunk 1]
```diff
@@ -293,3 +293,20 @@ include::{snippets}/should_document_get_usedbyresource_for_components/curl-reque
 
 ===== Example response
 include::{snippets}/should_document_get_usedbyresource_for_components/http-response.adoc[]
+
+[[resources-mysubscriptions-components]]
+==== Listing mysubscriptions components with fields
+
+A `GET` request will list all of the service's recent components.
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/components.adoc:300]: Add proper description related to new endpoint.

Comment (nikkuma7) [reply rest/resource-server/src/docs/asciidoc/components.adoc]: done.

File: rest/resource-server/src/docs/asciidoc/components.adoc [hunk 2]
```diff
@@ -293,3 +293,20 @@ include::{snippets}/should_document_get_usedbyresource_for_components/curl-reque
 
 ===== Example response
 include::{snippets}/should_document_get_usedbyresource_for_components/http-response.adoc[]
+
+[[resources-mysubscriptions-components]]
+==== Listing mysubscriptions components with fields
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/components.adoc:298]: remove -with fields as this endpoint uses no fields.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java [hunk 3]
```diff
@@ -177,6 +177,21 @@ public ResponseEntity<EntityModel<Component>> getComponent(
         return new ResponseEntity<>(userHalResource, HttpStatus.OK);
     }
 
+    @RequestMapping(value = COMPONENTS_URL + "/mysubscriptions", method = RequestMethod.GET)
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java:180]: use proper camel casing for new endpoint and make it as mySubscriptions

**Comment by ag4ums** [comment]
@nikkuma7 please rebase.

**Pull request merged**